app/controllers/api/v1/items/search_controller.rb

```diff
@@ -1,27 +1,39 @@
 class Api::V1::Items::SearchController < ApplicationController
+  def index
+    if params[:description]
+      item = Item.where('description ILIKE ?', "%#{params[:description].downcase}%")
+    elsif params[:unit_price]
+      item = Item.where(unit_price: params[:unit_price])
+    elsif params[:created_at]
+      date = Date.parse(params[:created_at])
+      item = Item.where(created_at: date.beginning_of_day..date.end_of_day)
+    elsif params[:updated_at]
+      date = Date.parse(params[:updated_at])
+      item = Item.where(updated_at: date.beginning_of_day..date.end_of_day)
+    else #params[:name]
+      item = Item.where('name ILIKE ?', "%#{params[:name].downcase}%")
+    end
+
+    return nil if item.empty?
+    render json: ItemSerializer.new(item)
+  end
```

Comment:
Refactored slightly by removing redundant code in each case. Will still need more refactoring to lighten up controller.